tests/modules/apigee_mock_client/src/MockHandlerStack.php

```diff
@@ -0,0 +1,164 @@
+   *
+   * @return $this
+   *
+   * @throws \Twig_Error_Loader
+   * @throws \Twig_Error_Runtime
+   * @throws \Twig_Error_Syntax
+   */
+  public function queueFromResponseFile($response_ids) {
+    $org_name = \Drupal::service('apigee_edge.sdk_connector')->getOrganization();
+
+    if (empty($this->responses)) {
+      // Get the module path for this module.
+      $module_path = \Drupal::moduleHandler()->getModule('apigee_mock_client')->getPath();
+      $this->responses = Yaml::parseFile($module_path . '/response_catalog.yml')['responses'];
+    }
+    // Loop through responses and add each one.
+    foreach ((array) $response_ids as $index => $item) {
+      // The catalog id should either be the item itself or the keys if an associative array has been passed.
+      $id = !is_array($item) ? $item : $index;
+      // Body text can have elements replaced in it for certain values. See: http://php.net/strtr
+      $context = is_array($item) ? $item : [];
+      $context['org_name'] = isset($context['org_name']) ? $context['org_name'] : $org_name;
+
+      // Add the default headers if headers aren't defined in the response catalog.
+      $headers = isset($this->responses[$id]['headers']) ? $this->responses[$id]['headers'] : [
+        'content-type' => 'application/json;charset=utf-8'
+      ];
+      // Set the default status code.
+      $status_code = !empty($this->responses[$id]['status_code']) ? $this->responses[$id]['status_code'] : 200;
+      $status_code = !empty($context['status_code']) ? $context['status_code'] : $status_code;
+
+      // Render the response content.
+      $template = str_replace('_', '-', $id) . '.json.twig';
+      $content = $this->twig->getLoader()->exists($template) ? $this->twig->render($template, $context) : '';
+
+      // Make replacements inside the response body and append the response.
+      $this->append(new Response(
+        $status_code,
+        $headers,
+        $content
+      ));
+    }
+
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __invoke(RequestInterface $request, array $options) {
+    try {
+      // Grab an item from the database queue and append it to the in-memory queue.
+      if ($item = $this->database_queue->claimItem()) {
+        parent::append(new Response(
+          $item->data['status'],
+          $item->data['headers'],
+          $item->data['body']
+        ));
+        $this->database_queue->deleteItem($item);
+      }
+
+      return parent::__invoke($request, $options);
+    } catch (\Exception $ex) {
+      // Fake a client error so the error gets reported during testing. The
+      // `apigee_edge_user_presave` is catching errors in a way that this error
+      // never gets reported to PHPUnit. This won't work for all use cases.
+      //
+      // @todo: Find another way to ensure an exception gets registered with PHPUnit.
+      throw new ClientErrorException(new Response(500, [
```

Comment:
Try to call `\PHPUnit_Framework_Assert::fail()` here.

Reply:
Great suggestion @mxr576.  Let’s put this in as a task to validate the approach.

Reply:
I guess you mean `throw new \PHPUnit_Framework_AssertionFailedError();`. It looks like `\PHPUnit_Framework_Assert::fail()` was removed.

Reply:
No, what I tried to achieve here that instead of throwing an exception, call PHPUnit's `fail()` method to fail a test and PHPUnit will do what it is necessary to do that.

Reply:
@mxr576 can you put a task in the backlog to address this? I think we need to get this pull request in.